merlin/models/tf/data_augmentation/negative_sampling.py

```diff
@@ -46,6 +47,11 @@ def call(self, inputs: TabularData, targets=None) -> Union[Prediction, TabularDa
             fist_input.shape[0] if not isinstance(fist_input, tuple) else fist_input[1].shape[0]
         )
 
+        if batch_size is None:
```

Comment:
If batch dimension is `None`, there's not much we can do here

Reply:
This was raising an exception when calling model.fit on a dataset at some stage in the calls made to the layer